From 4afff71046dceb0c7fa2a16e97125b17748e39ca Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 22 Mar 2021 12:49:06 +0200
Subject: [PATCH] [SW-0] gaudi2_arc_gnu: update psoc security key indices

A recent patch in the specs folder changed the names
of the locations where the keys are to be stored in
PSOC_SECURITY. This caused a compile error in arcboot.

To fix this, update to the correct names.

Below is the patch in specs that changed the naming -

    [SW-0] gaudi2: update specs (mostly java files)
    [gaudi2][rtl_v2.15][updated bmon in asic_trace bmon.java files][]

Change-Id: I24803944c5b6016fa9dd7ad77431c353d00b8db7
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2_arc_gnu/security.c b/gaudi2_arc_gnu/security.c
index 77e1da9..86584bd 100644
--- a/gaudi2_arc_gnu/security.c
+++ b/gaudi2_arc_gnu/security.c
@@ -491,11 +491,11 @@
 	uint8_t max_dwords_stored = NUM_DWORDS_AVLBL_IN_PSOC_SEC, i, j;
 	uint32_t key_soc_location_base;
 	uint32_t key_address[] = {
+		mmPSOC_SECURITY_HABANA_KEY_0,
 		mmPSOC_SECURITY_PUBLIC_KEY_0_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_0_8,
-		mmPSOC_SECURITY_PUBLIC_KEY_1_4,
+		mmPSOC_SECURITY_PUBLIC_KEY_1_0,
 		mmPSOC_SECURITY_PUBLIC_KEY_2_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_2_8
+		mmPSOC_SECURITY_PUBLIC_KEY_3_0
 	};
 
 	for (i = 0; i < max_keys; i++) {
